external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala

```diff
@@ -64,7 +64,7 @@ private[avro] class AvroFileFormat extends FileFormat with DataSourceRegister {
     // Schema evolution is not supported yet. Here we only pick a single random sample file to
     // figure out the schema of the whole dataset.
     val sampleFile =
-      if (conf.getBoolean(AvroFileFormat.IgnoreFilesWithoutExtensionProperty, true)) {
+      if (AvroFileFormat.ignoreFilesWithoutExtensions(conf)) {
```

Comment:
I tried running queries. The option `avro.mapred.ignore.inputs.without.extension` is not set in `conf`.  This is a bug in `spark-avro`.
Please read the value from `options`. It would be good to have a new test case with `avro.mapred.ignore.inputs.without.extension` as true.

Reply:
The `avro.mapred.ignore.inputs.without.extension` is hadoop's parameter. This PR aims to change the default behavior only. I don't want to convert the hadoop parameter to Avro datasource option here.

Reply:
Here is how people use the option so far: https://github.com/databricks/spark-avro/issues/71#issuecomment-280844562 . Probably we should seperatly from the PR discuss how we are going to fix the "bug" and break backward compatibily.

Reply:
The Hadoop config can be changed like:
```scala
spark
  .sqlContext
  .sparkContext
  .hadoopConfiguration
  .set("avro.mapred.ignore.inputs.without.extension", "true")
```

Reply:
Can we submit a separate PR to add a new option for AVRO? We should not rely on hadoopConf to control the behaviors of AVRO.

Reply:
Here is the PR: https://github.com/apache/spark/pull/21798 Please, have a look at it.